doc/source/programs/gdal_raster_proximity.rst

```diff
@@ -0,0 +1,90 @@
+.. _gdal_raster_proximity:
+
+================================================================================
+``gdal raster proximity``
+================================================================================
+
+.. versionadded:: 3.11
+
+.. only:: html
+
+    Produces a raster proximity map.
+
+.. Index:: gdal raster proximity
+
+Synopsis
+--------
+
+.. program-output:: gdal raster proximity --help-doc
+
+Description
+-----------
+
+:program:`gdal raster proximity` generates a raster proximity map indicating the distance from
+the center of each pixel to the center of the nearest pixel identified as a target pixel.
+Target pixels are those in the source raster for which the raster pixel value is in the set of
+target pixel values.
+
+Standard options
+++++++++++++++++
+
+.. include:: gdal_options/of_raster_create_copy.rst
+
+.. include:: gdal_options/co.rst
+.. include:: gdal_options/overwrite.rst
+
+.. include:: gdal_options/ot.rst
+
+.. option:: -b, --band <BAND>
+
+    Input band (1-based index)
+
+.. option:: --target-values <TARGET-VALUES>
+
+    A single value or a comma separated list of target pixel values in the source image
+    to be considered target pixels.
+    If not specified, all non-zero pixels will be considered target pixels.
+
+.. option:: --max-distance <MAX-DISTANCE>
+
+    Maximum distance. The nodata value will be used for pixels beyond this distance.
+    Distance is interpreted in pixels unless `--distance-units geo` is specified.
+
+.. option:: --distance-units <pixel|geo>
+
+    Units for the distance. If `geo` is specified, the distance will be calculated in
+    georeferenced units (meters or degrees) using the pixel size of the input raster.
+    Otherwise, the distance is interpreted in pixels.
+
+.. option:: --fixed-buffer <FIXED-BUFFER>
+
+    Define a fixed value to be written in the output pixels that are within the
+    specified maximum distance from the target pixels, instead of the actual distance.
+
+.. option:: --nodata <NODATA>
+
+    Nodata value for the output raster. If not specified, the nodata value of the input band will be used.
```

Comment:
What happens if the input band does not have a NoData value?

Reply:
According to https://gdal.org/en/stable/programs/gdal_proximity.html 65536 is used but let me write a few tests for that: I doubt that it will work with `Byte` output type.

Reply:
Tests added and documentation expanded.